status-im/status-react
# Revamp legacy Privacy and security settings

Fixes https://github.com/status-im/status-mobile/issues/20618

### Summary

This PR revamps the legacy _Privacy and security_ screen to meet the new designs.

Note that we (devs, designers, and QAs) agreed off-GitHub that the following changes are desirable. See in [Figma](https://www.figma.com/design/JlpPhJp0SMnEyBQy4nOZHo/Settings-for-Mobile?node-id=7845-50334&t=MvuqAoxmsqcnqTVx-0)

- [x] Created function `utils.navigation/navigate-back`, but I only used this in 2 namespaces to keep the PR under scope.
- [x] Removed setting `Set dApp access permissions` (won't be used anymore).
- [x] Removed setting `Display collectibles` (won't be used anymore).
- [x] Removed setting `Chat link previews setting` (they already had no effect).
- [x] Removed `Reset password setting`. This setting is already supported in `Profile > Password > Change password`.
- [x] Removed `Delete my profile`. Profile deletion is already supported in the login screen.
- [x] Moved setting `Show your profile picture` to `Profile > Privacy and security`.
- [x] Moved setting `Accept new chats from <XYZ>` to `Allow new contact requests` in `Profile > Messages`. Now it's just a toggle.
- [x] Moved Block screenshots (Android) or Hide Preview (iOS) which are both accessible in `Profile > Privacy and security`.
      
[demo-pr.webm](https://github.com/status-im/status-mobile/assets/46027/cb492954-f913-4bc9-b2ef-000278aa8d11)

[block screenshot.webm](https://github.com/status-im/status-mobile/assets/46027/1c488cf4-f069-42c1-820f-e97c0b7d5cc9)

### Review notes

I removed most dead code, all in a single commit to help you review.

#### Platforms

- Android
- iOS

### Areas that may be impacted

Old legacy settings that are still relevant.

### Steps to test

This PR is not changing how the legacy settings work under the hood. It's mostly lipstick and a lot of code removal, but the code removed was very clearly not used after I removed the legacy settings, so I see this PR as low risk, but requiring QA.

Recommended to check if the settings described above work as expected.

status: ready
